### PR TITLE
fix: add StatementBlock to support local assignments in blocks and re-enable the scope.lua test

### PR DESCRIPTION
**Pull request by sfluor**

### CODE REVIEW

File: src/main/java/jua/objects/LuaFunction.java [hunk 1]
```diff
@@ -18,18 +18,18 @@ public LuaFunction(ArrayList<String> argNames, Scope environment, StatementList
...
       return (LuaFunction) o;
     }
     throw new IllegalCastException(String.format("%s is not a function", o.repr()));
   }
 
+  @Override
+  public String repr() {
+    return String.format("function(%s) %s", argNames, block);
+  }
+
```
Comment (remicalixte) [inline src/main/java/jua/objects/LuaFunction.java]: Unrelated changes

**Comment by remicalixte** [comment]
Otherwise looks good to me

File: src/main/java/jua/objects/LuaFunction.java (hunk 1 above)
Comment (sfluor) [reply src/main/java/jua/objects/LuaFunction.java]: it's the formatting script :/
